### ISSUE TITLE
TODO: Plains of Snow Total Rework

### ISSUE THREAD

**Issue by VictoriumWoW** [The-Alpha-Project/alpha-core#1457]
The Plains of Snow map has two bindstones. One is within an area with orc banners, the other in an area with dwarven braziers. 

They can be seen quite clearly here: https://youtu.be/2z2w2JFoPnY?t=70

There are four NPCs with moderately low IDs, all of whom are side-by-side and completely unused.

![FieldsOfSnowNPCs](https://github.com/user-attachments/assets/da16dd55-fff8-4033-a220-828a68868ca4)

There are two binders. Both have frost-related themes in their names (Frostbane and Iceflow), while one is an orc and the other is a human. Similarly, there are two "Battleguard" NPCs, one of whom is an orc, the other a human. All of these NPCs use placeholder models.

Given there's one Battleguard and Binder for each faction, the frost-related themes in the names of the Binders, and the fact there's one bindstone in each faction's area in Plains of Snow, these NPCs seem to have appeared in Plains of Snow. My theory is that Alliance players would be binded to Baelish Frostbane, located at the Alliance area's bindstone, while Horde players would be binded to Orgrun Iceflow, located at the Horde area's bindstone, with their binds likely resetting to the ones they had before entering the map upon leaving it. The Alliance Battleguard would be near the Alliance bindstone, likely at the end of the path in front of it, while the Horde Battleguard would be near the Horde bindstone, likely at the end of the path in front of it. Functionally, it probably worked the same as AV, where if the Alliance Battleguard died the Horde would win while if the Horde Battleguard died the Alliance would win. Winning probably resulted in just a popup, given the early state of the battleground at this time (example from live servers: https://i.imgur.com/4WMQRsW.jpeg).

**Changes Needed**

- [ ] Spawn the Alliance Battleguard (ID 2279) at the end of the path at the Alliance side.
- [ ] Alliance Battleguard should be level 70 (the planned max level at this stage of development) and should be assigned to faction ID 11 (Human, Stormwind). On death, a generic message (i.e. yelling "Horde won!") should appear.
- [ ] Spawn the Horde Battleguard (ID 2280) at the end of the path at the Horde side.
- [ ] Horde Battleguard should be level 70 (the planned max level at this stage of development) and should be assigned to faction ID 85 (Horde, Orgrimmar). On death, a generic message (i.e. yelling "Alliance won!") should appear.
- [ ] Spawn Baelish Frostbane (ID 2281) in front of the bindstone at the Alliance side.
- [ ] Baelish Frostbane's Binder functionality should work the same as regular Binders, except Alliance players should only be bound to him while inside of the Plains of Snow map.
- [ ] Spawn Orgrun Iceflow (ID 2282) in front of the bindstone at the Horde side.
- [ ] Orgrun Iceflow's Binder functionality should work the same as regular Binders, except Horde players should only be bound to him while inside of the Plains of Snow map.

**Comment by geo-tp** [comment]
could be related

![image](https://github.com/user-attachments/assets/56eee38b-5d5f-4bf8-b5f9-d7cbb2cc5cbb)

GameObjectDisplayInfo
![towers](https://github.com/user-attachments/assets/b9348557-4e61-47b0-aca7-7464ed5acfe4)

Gameobject
![image](https://github.com/user-attachments/assets/e1d1f5ac-cc37-4d12-9e63-6776511d5e4b)

Plains of Snow in Warcraft 2
https://www.youtube.com/watch?v=wUxgkLYwbnw

**Comment by VictoriumWoW** [comment]
I'm not too sure about the mines, but there are many dead ends in the BG where towers could've been placed. Blizz was considering siege equipment during this time so they may have been buildable/destructive, with the battleguards spawning around them.

**Comment by VictoriumWoW** [comment]
If we look at it under this lens, then considering that this battleground didn't seem to have any objectives during this time, it was likely just a pvp zone at this time that served as a "template" for other battlegrounds like Kalidar, which has a lot of leftover stuff tied to it.

**Comment by geo-tp** [comment]
I tested both towers, and considering the markings on the ground, there are 2 locations that could most likely match
![allytower](https://github.com/user-attachments/assets/2703f21b-ec20-41d1-8363-245b112f148e)
![hordetower](https://github.com/user-attachments/assets/8a97a56e-ff8b-4cda-a2c1-c9a9b8400068)

Still looking at the marks on the ground, there is probably something that should be spawned in the center of the map

**Comment by VictoriumWoW** [comment]
These towers do match suspiciously well into the map. My concerns mainly revolve around their functionality, as it's very uncommon for Blizz to add large structures like these in the form of gameobjects. As for the centre of the map, that's a total mystery. Could just be leftovers from when the map was copied in from pre-0.5 Dun Morogh, although I did notice that Grimtooth, a rare in Alterac Valley, exists in 0.5.3 and has a missing entry next to him which could be related. https://db.thealphaproject.eu/index.php?action=show_creature&id=603&filter=grimtooth&sort_order=name&pos=1&max=1

**Comment by geo-tp** [comment]
Grimtooth seems to have a really low entry to belong to Plains of Snow, it was most likely unused for long and then reused in Alterac but it could be possible

Yes, there are very few WMO models, most likely if these two towers do indeed belong to Plains of Snow, then the goldmine probably should as well

#### GameobjectDisplayInfo wmo
![wmos](https://github.com/user-attachments/assets/4260cabe-8a2a-4414-be04-0ddcaf4c1e01)

Using these models server side rather than client side could indicate that they should have the ability to appear/disappear, or be built/destroyed

Reading the link above from crawler suggests that battlegrounds might look like sort of RTS. Plains of Snow is also a Warcraft 2 map, so it's a port from an RTS game. 

Build/Destroy system could explain why these towers are spawned server side

#### Items (show the connection between Mine/Tower and PVP)
![items](https://github.com/user-attachments/assets/71f9deb7-d9c1-4e7b-b4c0-b34c71f10122)

#### Creatures (most likely linked to items)
![range](https://github.com/user-attachments/assets/e7197678-aa7b-4742-aec2-331beb4d004a)


Note that the creature entries are much higher, which could be explained by an initial push of the binders/battleguards, and then the implementation of the tower/mine system. This high ids could also be related to Kalidar, but that would mean the entire system would be connected to Kalidar instead of Plains of Snow

**Comment by VictoriumWoW** [comment]
Those high ID NPCs seem to line up more with Kalidar, as Kalidar has a lot of tribal tents and entire mines baked into the map, with NPCs corresponding to them (ogre Caravan Guards, bandit Mine Bosses, etc). As for the deeds, they're classified as keys, so they were probably used to unlock a gameobject which created the tower instead of any sort of spell to place in the tower like the player houses.

**Comment by geo-tp** [comment]
Given the shape of the terrain in Kalidar, one could imagine that it's so flat to allow for the possibility of spawning structures in a manner similar to Create House

https://github.com/user-attachments/assets/d7e4fe94-d2cd-4b47-83e5-7cc521d7e597

![Kalidar_map](https://github.com/user-attachments/assets/5a9a305b-b4c1-4f68-ac59-8a0835b2dd19)

It seems strange that Kalidar is entirely shaped but that the terrain remains so flat, this could be explained in this way

Plains of Snow is fully shaped, including the ground, so this spawning system wouldn’t be adaptable to this map. Here, we can see that Gameobject Locks require Horde/Alliance PvP items to unlock them

#### Lock.dbc
![image](https://github.com/user-attachments/assets/dac65936-ded9-437b-8ef0-4377ec74e4a5)

#### items
![items](https://github.com/user-attachments/assets/7a21262b-e13f-4d06-955c-39fe3b24bb8f)

**Comment by VictoriumWoW** [comment]
Seems like there may have been plots of land sprinkled around the map where Alliance and Horde could use deeds to spawn the towers/mines. In consideration to the pre-built mines and towers, we have data for enemy mine bosses so those probably populated the pre-built mines while the tower may have worked similarly to something like the Snowfall graveyard, where Alliance and Horde would contest for its control.

**Comment by geo-tp** [comment]
It's possible that Kalidar was abandoned early on, even before version 0.5.3

The fact that Kalidar has no bindstone might indicate that it was created even before this system was implemented, especially considering that it also has a very low map id (Kalidar is 17, for reference, PVPZone01 is 30, and Deadmines is 36)

Kalidar will never have subzone text or bindstones, even in later versions 

#### Kalidar 0.6 (No subzone, no bindstones)
![kalidar06](https://github.com/user-attachments/assets/bf8ba209-7e7d-4c5b-a8e4-657e92b58da9)


#### Removed from the game in 0.7 (same loc as 0.6)
![kalidar07](https://github.com/user-attachments/assets/05e1a7cd-9a2b-42c8-9d5e-00a6d901c17e)

In contrast, Plains of Snow already has bindstones and even binders, it has subzone text, and we can see that in version 0.6, Alliance/Horde markers were added to the map

#### Markers in 0.6
![markers](https://github.com/user-attachments/assets/581eeb5a-deb2-44ea-975c-ad45369bab07)


#### Plains of Snow still there in 0.8
![pos08](https://github.com/user-attachments/assets/d79da2d1-29a4-4e16-8f24-c0c5cf5b758b)

Given that the creatures have very high IDs, they were likely added in version 0.5.3 It's possible that the game objects, items, and NPCs might belong to Plains of Snow instead of Kalidar if this map was already abandonned at this time.

#### High Ids most likely added in 0.5.3
![image](https://github.com/user-attachments/assets/193d87d8-172b-4694-a82b-b5d84f90f328)

**Comment by VictoriumWoW** [comment]
Interesting. I'm not sure where the caravans would've come from since there's no camps in Plains of Snow, perhaps they just appeared from the borders or center. That's also probably where you got the deeds to towers and mines (when you captured them.) There may be issues with spawning the mines using the "Dungeon" model, since there's no holes in the terrain for players to pass through at spots where mines could be.

**Comment by geo-tp** [comment]
GOs related to this 

https://www.wowhead.com/classic/object=28024/caravan-chest
![caravanchest](https://github.com/user-attachments/assets/972f732b-804f-44d8-b5ff-2a4f94dcb176)

https://www.wowhead.com/classic/object=28025/pvp-tower-supplies
![pvptowersupplies](https://github.com/user-attachments/assets/a14fc74e-6248-4de0-a983-280bc56d74fc)

PVP Tower Supplies is the original gameobject used by Lock id 199  with 

#### Items
![image](https://github.com/user-attachments/assets/06f6e1e7-e66c-4899-9b91-4ec474341d2a)

#### Lock.dbc
![image](https://github.com/user-attachments/assets/ab25c18e-38a1-4f79-8883-193ccc5e404c)

**Comment by geo-tp** [comment]
Here's the originals towers and mine gameobjects_template and models

#### Orc
![image](https://github.com/user-attachments/assets/f2b09e5e-0c85-4af7-9c8f-6e97d08ea795)
![image](https://github.com/user-attachments/assets/9c56a8ed-65c3-4cb4-86b6-9c82cd6ad89d)

#### Human
![image](https://github.com/user-attachments/assets/696dfd5a-4cb4-4e62-87ef-340650169fbd)
![661](https://github.com/user-attachments/assets/3a0eab73-fde2-4323-91b3-13a540649118)


##### Mine
![image](https://github.com/user-attachments/assets/aab6a83b-1fec-4cec-b960-9e6ec88acb0f)
![image](https://github.com/user-attachments/assets/b8aa78b7-bb23-4436-9de9-5ea4332a6ce8)


The WMO were mostly related to Kalidar

**Comment by VictoriumWoW** [comment]
Looks reasonable enough. The goldmine being a shed model is a real teller that these towers and such were probably spawned into the map, and if those gameobjects are activated by the deeds it would make sense to add a script that spawns them in on-use. Because the Goldmine is activated by the deed, it's likely that it was guarded by the enemy Mine Boss and Guards first, then when they died it could be unlocked. Caravans probably provided the deeds, which like the mines had to be captured by killing the ogres.

**Comment by VictoriumWoW** [comment]
I think the PH NPCs and these PVP gameobjects may have been used for Plains of Snow, but the lack of information is going to lead to us adlibing a lot of stuff. The only NPCs we know for certain were in Plains of Snow were the Battleguards and Binders, the rest could've been in one of the other PVP maps.

**Comment by VictoriumWoW** [comment]
It's been a while since anyone's posted in here, so I felt like adding something now that Plains of Snow is being used as the PVP zone when /pvp is toggled. Considering that the Binders existed in Plains of Snow, it's likely that the Battleguards were literally just guard NPCs that protected the spawn points in order to prevent camping.

These (albeit sketchy) WOTLK server databases appear to include TrinityCore sniffs of their data. We can see their factions, levels, and health/damage stats here:
https://woehead.way-of-elendil.fr/?npc=2279
https://woehead.way-of-elendil.fr/?npc=2280

In summary:
- Both Battleguards were level 72
- The Alliance Battleguard was apart of faction ID 189 (Alliance Generic)
- The Horde Battleguard was apart of faction ID 66 (Horde Generic, called Horde Friendly in 0.5.3)

We can probably spawn these guys around the Alliance and Horde Binders, maybe four or close to the binders in a reasonable location.

**Comment by VictoriumWoW** [comment]
To add to this, these same websites list those binders as being level 62. Since Plains of Snow existed in 0.8 with its map markers still intact, it's possible they scaled these NPCs up in levels between 0.6 and 0.8 when the level cap was surely going to be 60. In 0.5.3, these Battleguards may have been level 55 (the Binder's levels +10).
